test/unit_tests/arkode/C_serial/ark_test_erkstepsetforcing.c

```diff
@@ -0,0 +1,386 @@
+      return (1);
+    }
+  }
+
+  if (argc > 3)
+  {
+    if (argc < 5)
+    {
+      printf("ERROR: Both the initial and final time are required\n");
+      return (1);
+    }
+    T0 = (realtype)atof(argv[3]);
+    Tf = (realtype)atof(argv[4]);
+    if (SUNRabs(T0) >= SUNRabs(Tf))
+    {
+      printf("ERROR: |T0| must be less than |Tf|\n");
+      return (1);
+    }
+  }
+
+  if (argc > 5)
+  {
+    if (argc < 7)
+    {
+      printf("ERROR: Both tshift and tscale are required\n");
+      return (1);
+    }
+    tshift = (realtype)atof(argv[5]);
+    tscale = (realtype)atof(argv[6]);
+    if (SUNRabs(tscale) < TINY)
+    {
+      printf("ERROR: |tscale| must be greater than %" GSYM "\n", TINY);
+      return (1);
+    }
+  }
+
+  /* Output test setup */
+  printf("\nerkStep_SetInnerForcing unit test:\n");
+  printf("   NEQ    = %li\n", (long int)NEQ);
+  printf("   order  = %i\n", order);
+  printf("   t0     = %.1" GSYM "\n", T0);
+  printf("   tf     = %.1" GSYM "\n", Tf);
+  printf("   tshift = %.1" GSYM "\n", tshift);
+  printf("   tscale = %.1" GSYM "\n", tscale);
+  printf("   reltol = %.1" ESYM "\n", reltol);
+  printf("   abstol = %.1" ESYM "\n\n", abstol);
+
+  /* Create the SUNDIALS context object for this simulation. */
+  SUNContext_Create(NULL, &sunctx);
+
+  /* Create solution vector and initialize to zero */
+  y = N_VNew_Serial(NEQ, sunctx);
+  if (check_flag((void*)y, "N_VNew_Serial", 0)) return 1;
+
+  ans = N_VClone(y);
+  if (check_flag((void*)ans, "N_VClone", 0)) return 1;
+
+  tmp = N_VClone(y);
+  if (check_flag((void*)tmp, "N_VClone", 0)) return 1;
+
+  /* allocate vector array for polynomial forcing */
+  forcing = N_VCloneVectorArray(order + 1, y);
+  if (check_flag((void*)forcing, "N_VCloneVectorArray", 0)) return 1;
+  for (i = 0; i < order + 1; i++)
+  {
+    if (check_flag((void*)forcing[i], "N_VCloneVectorArray", 0)) return 1;
+  }
+
+  /* fill forcing vectors with random data */
+  for (i = 0; i < order + 1; i++)
+  {
+    data = N_VGetArrayPointer(forcing[i]);
+    for (j = 0; j < NEQ; j++)
+    {
+      data[j] = (realtype)rand() / (realtype)RAND_MAX;
+    }
+  }
+
+  /* compute the true solution */
+  flag = compute_ans(Tf, tshift, tscale, forcing, order, ans);
+  if (check_flag(&flag, "compute_ans", 1)) return 1;
+
+  printf("True solution:\n");
+  N_VPrint_Serial(ans);
+
+  /* ---------------------------------------------------------------------------
+   * explicit test
+   * -------------------------------------------------------------------------*/
+
+  /* initialize solution vector */
+  flag = compute_ans(T0, tshift, tscale, forcing, order, y);
+  if (check_flag(&flag, "compute_ans", 1)) return 1;
+
+  /* Create ARKode mem structure */
+  arkode_mem = ERKStepCreate(f, T0, y, sunctx);
+  if (check_flag((void*)arkode_mem, "ERKStepCreate", 0)) return 1;
+
+  /* Specify tolerances */
+  flag = ERKStepSStolerances(arkode_mem, reltol, abstol);
```

Comment:
Should these and other functions use the `ARKode*` version?

Reply:
Yes; I wrote this about 12 months before we consolidated the API, and apparently forgot to update this file.